### ISSUE TITLE
listAll() -> Xcode warning: Thread running at QOS_CLASS_USER_INTERACTIVE waiting on a lower QoS thread

### ISSUE THREAD

**Issue by Kondamon** [firebase/firebase-ios-sdk#10698]
### Description

After invoking `listAll()` of FirebaseStorage I get the warning: `.../gtm-session-fetcher/Sources/Core/GTMSessionFetcherService.m:307 Thread running at QOS_CLASS_USER_INTERACTIVE waiting on a lower QoS thread running at QOS_CLASS_DEFAULT. Investigate ways to avoid priority inversions`.

I want to see no warning.

### Reproducing the issue

1) Enforce AppCheck for Firebase Storage in Firebase console

2)
```
let storage = Storage.storage()
let ref = storage.reference(withPath: .path)
let musicStorageList = try await ref.listAll()
```

### Firebase SDK Version

10.3.0

### Xcode Version

14.2

### Installation Method

Swift Package Manager

### Firebase Product(s)

App Check, Crashlytics, Firestore, Remote Config, Storage

### Targeted Platforms

iOS

### Relevant Log Output

```shell
StackTrace:
#0	0x0000000103fe933c in -[GTMSessionFetcherService sessionForFetcherCreation] at /SourcePackages/checkouts/gtm-session-fetcher/Sources/Core/GTMSessionFetcherService.m:307
#1	0x0000000103fcbcf4 in -[GTMSessionFetcher beginFetchMayDelay:mayAuthorize:mayDecorate:] at /SourcePackages/checkouts/gtm-session-fetcher/Sources/Core/GTMSessionFetcher.m:653
#2	0x0000000103fd29c8 in -[GTMSessionFetcher authorizer:request:finishedWithError:] at /SourcePackages/checkouts/gtm-session-fetcher/Sources/Core/GTMSessionFetcher.m:1727
#3	0x0000000103fd27b0 in __37-[GTMSessionFetcher authorizeRequest]_block_invoke at /SourcePackages/checkouts/gtm-session-fetcher/Sources/Core/GTMSessionFetcher.m:1694
#5	0x0000000103f91f18 in closure #3 in StorageTokenAuthorizer.authorizeRequest(_:completionHandler:) at /SourcePackages/checkouts/firebase-ios-sdk/FirebaseStorage/Sources/Internal/StorageTokenAuthorizer.swift:78
```

**Comment by Kondamon** [comment]
Is resolved with GTMSessionFetcher 3.1.0.

**Comment by paulb777** [comment]
Yep, duplicate of https://github.com/firebase/firebase-ios-sdk/issues/10476

**Comment by Kondamon** [comment]
@paulb777, the issue was that updating to Firebase 10.3.0 didn't update GTMSessionFetcher to the latest 3.1.0 but kept the latest 2.x version. I had to manually update to 3.1.0 by explicitly adding `https://github.com/google/gtm-session-fetcher.git` and specifying the version range, then it worked.

This is caused by this dependency in Firebase 10.3.0.
```
.package(
      name: "GTMSessionFetcher",
      url: "https://github.com/google/gtm-session-fetcher.git",
      "2.1.0" ..< "4.0.0"
    ),
```

**Comment by paulb777** [comment]
The alternative in Xcode would be to do File->Packages->Update to Latest Versions
